Assets/SO Architecture/Editor/ClampValueHelper.cs

```diff
@@ -0,0 +1,115 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Reflection;
+using UnityEditor;
+using UnityEngine;
+
+#if ODIN_INSPECTOR
+using Sirenix.Utilities.Editor;
+#endif
+
+namespace ScriptableObjectArchitecture.Editor
+{
+    /// <summary>
+    /// A helper class for drawing better controls for clamped value types (mostly simple C# types)
+    /// </summary>
+    public static class ClampValueHelper
+    {
+        private static readonly List<Type> SUPPORTED_INT_TYPES;
+        private static readonly List<Type> SUPPORTED_FLOAT_TYPES;
+
+        static ClampValueHelper()
+        {
+            SUPPORTED_INT_TYPES = new List<Type>
+            {
+                typeof(sbyte),
+                typeof(byte),
+                typeof(short),
+                typeof(ushort),
+                typeof(int)
+            };
+
+            SUPPORTED_FLOAT_TYPES = new List<Type>
+            {
+                typeof(float)
+            };
+        }
+
+        /// <summary>
+        ///  Draws a slider for the passed <see cref="SerializedProperty"/> values.
+        /// </summary>
+        /// <param name="valueProp"></param>
+        /// <param name="minProp"></param>
+        /// <param name="maxProp"></param>
+        public static void DrawClampRangeLayout(
+            SerializedProperty valueProp,
+            SerializedProperty minProp,
+            SerializedProperty maxProp)
+        {
+            var valueType = GetType(valueProp);
+
+            if (SUPPORTED_INT_TYPES.Contains(valueType))

```

Comment:
Doesn't the calling method also check whether the type is supported?

Reply:
It's different; from above this point in the stack `CanDrawClampRange` is called which returns true if the value is a supported `float` OR `int` type. Here, we're checking if the passed value type is a supported `int` or `float` type specifically so we can either draw the int or float Odin GUI field (`SirenixEditorFields.RangeIntField` or `SirenixEditorFields.RangeFloatField`) along with the relevant serialized property `intValue` or `floatValue` member.